Tests/scripts/configure_tests.py

```diff
@@ -445,7 +462,47 @@ def collect_changed_ids(integration_ids, playbook_names, script_names, modified_
     if deprecated_commands_message:
         print_color(deprecated_commands_message, LOG_COLORS.YELLOW)
 
-    return tests_set, catched_scripts, catched_playbooks
+    packs_to_install = set()
+
+    id_set_integrations = id_set.get('integrations', [])
+    for integration in id_set_integrations:
+        integration_id = list(integration.keys())[0]
+        integration_object = integration[integration_id]
+        if integration_id in integration_ids:
+            integration_pack = integration_object.get('pack')
+            if integration_pack:
+                print('Found integration {0} in pack {1} - adding to packs to install'.format(
+                    integration_id, integration_pack))
+                packs_to_install.add(integration_object.get('pack'))
+            else:
+                print('Found integration {0} without pack - not adding to packs to install'.format(integration_id))
+
+    id_set_playbooks = id_set.get('playbooks', [])
+    for playbook in id_set_playbooks:
+        playbook_object = list(playbook.values())[0]
+        playbook_name = playbook_object.get('name')
+        if playbook_name in playbook_names:
+            playbook_pack = playbook_object.get('pack')
+            if playbook_pack:
+                print('Found playbook {0} in pack {1} - adding to packs to install'.format(
+                    playbook_name, playbook_pack))
+                packs_to_install.add(playbook_pack)
+            else:
+                print('Found playbook {0} without pack - not adding to packs to install'.format(playbook_name))
+
+    id_set_script = id_set.get('scripts', [])
+    for script in id_set_script:
+        script_id = list(script.keys())[0]
+        script_object = script[script_id]
+        if script_id in script_names:
+            script_pack = script_object.get('pack')
+            if script_pack:
+                print('Found script {0} in pack {1} - adding to packs to install'.format(script_id, script_pack))
+                packs_to_install.add(script_object.get('pack'))
+            else:
```

Comment:
another function please

Reply:
you mean a function to iterate the list of entities and extract the pack name?

Reply:
No, just extract this logic from this function, it's big as it is, no need to make it even bigger